src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -132,12 +133,10 @@ private Task OnProjectFactoryCompletedAsync()
         {
             var previousProjectContext = _currentAggregateProjectContext;
             var newProjectContext = await UpdateProjectContextAsync().ConfigureAwait(false);
+
             if (previousProjectContext != newProjectContext)
             {
-                // Dispose existing subscriptions.
-                DisposeAndClearSubscriptions();
```

Comment:
@natidea Could we run into similar issues for NuGet restorer? Due to handle duplicate updates?